feat: expose node internals via http for testing

glottologist opened the pull request:
**Describe the changes**
A clear and concise description of what the bug fix, feature, or improvement is.

**Related Issue(s)**
Please link to the issue(s) that will be closed with this PR.

**Checklist**

- [ ] Tests have been added/updated for the changes.
- [ ] Documentation has been updated for the changes (if applicable).
- [ ] The code follows Rust's style guidelines.

**Additional Context**
Add any other context about the pull request here.

DanMacDonald commented:
Good start, we need to make sure the peers are each making 3 pledges so their storage modules are assigned partition hashes and get assigned to ledger slots. Otherwise their storage modules remain unassigned and there's no data to sync.

DanMacDonald commented:
> Chain monitoring: Added /chain/height endpoint to report current blockchain height

This isn't strictly necessary as this information is available via the info endpoint.

https://github.com/Irys-xyz/irys/blob/63187735942d2795fb61595b13080adb7fba4e00/crates/api-server/src/routes/index.rs#L24-L37

DanMacDonald commented:
> Ledger summaries: Added /ledger/{ledger}/{node_id}/summary for monitoring of node assignment

We don't really have the concept of a `node_id` just miner address, I think in this case it's probably better to pass the parameter as a `web::Path<Address>` and let the framework handle the path param parsing rather than passing it as a string that we manually parse.  Also, lets call it `miner_address` for consistency, and not introduce `node_id`.

DanMacDonald reviewed (approved):
you cooked on this one, test looks good!

This was a bit difficult to review as your linter seemed to change all the format strings in any file you opened.  Hopefully less of that churn in the future.

glottologist commented:
> > Ledger summaries: Added /ledger/{ledger}/{node_id}/summary for monitoring of node assignment
> 
> We don't really have the concept of a `node_id` just miner address, I think in this case it's probably better to pass the parameter as a `web::Path<Address>` and let the framework handle the path param parsing rather than passing it as a string that we manually parse. Also, lets call it `miner_address` for consistency, and not introduce `node_id`.

Fixed with ae0952bea3c5af286046f45f8a56b96e16c2a95a
